### PR TITLE
Swap to pyproject.toml setup

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Modern Pythonic practice has moved away from `setup.py` +/- `.cfg` towards `pyproject.toml`

This updates the package accordingly

### CODE REVIEW

File: docs/.readthedocs.yaml
Comment (CodyCBakerPhD) [inline docs/.readthedocs.yaml:1]: I copied over these two RTD files to start a smooth transition to keeping the files there instead of the outer level to reduce clutter

It works fine as long as the settings on the RTD admin page are adjusted, which is why I will wait until after this PR is merged to do that and then remove the files still at the outer level (so it doesn't break on the other current PRs)

File: README.md
Comment (CodyCBakerPhD) [inline README.md:2]: This change should fix the issue with the logo not rendering on PyPI: https://pypi.org/project/nwbinspector/

File: README.md
Comment (CodyCBakerPhD) [inline README.md:20]: These are stylistic changes to the front page

File: pyproject.toml
Comment (CodyCBakerPhD) [inline pyproject.toml:33]: This also removes Python 3.8 support, which ends its security upgrades next month

**Comment by CodyCBakerPhD** [comment]
This will take a bit more work - will splinter a few items into smaller PRs

**Comment by CodyCBakerPhD** [comment]
@stephprince OK this is ready now

I'll remove the RTD files on the outer level in a followup so the docs have a smooth transition

File: pyproject.toml
Comment (stephprince) [inline pyproject.toml]: I think hdmf should support numpy >2.0 now

```suggestion
    "numpy>=1.22.0"
```

File: pyproject.toml
Comment (stephprince) [inline pyproject.toml]:
```suggestion
    "pynwb>=2.8",   # NWB Inspector should always be used with most recent minor versions of PyNWB
```

File: pyproject.toml
Comment (stephprince) [inline pyproject.toml:7]: not sure if the version was meant to be updated in this PR?

**Review by stephprince — APPROVED** [review]
Looks good to me. Added a couple small suggestions.

File: pyproject.toml
Comment (CodyCBakerPhD) [reply pyproject.toml]: Oh, this is good now? Thanks for checking

File: pyproject.toml
Comment (CodyCBakerPhD) [reply pyproject.toml:7]: It was. Just in case. (using a different build backend is a fundamental change; not per se one expected to break back compatibility, but it's fairly significant)

Maybe after the remaining big hard deprecations next March you might want to do an official v1.0.0 release

File: pyproject.toml
Comment (CodyCBakerPhD) [reply pyproject.toml]: Ah right, since that was merged first

Good catch

File: pyproject.toml
Comment (stephprince) [reply pyproject.toml:7]: That makes sense. And yes an official release after the big hard deprecations sounds like a good plan.

File: pyproject.toml
Comment (CodyCBakerPhD) [inline pyproject.toml]:
```suggestion
    "numpy>=1.22.0,<2.0.0" # TODO: add compatibility for 2.0
```

Ahh looks like we still have some 2.0 incompatibilities local to Inspector here: https://github.com/NeurodataWithoutBorders/nwbinspector/actions/runs/10907985074/job/30272984050?pr=507#step:8:414

Will deal with this in a follow-up

**Pull request merged**
